lib/rule-tester/rule-tester.js

```diff
@@ -416,21 +416,30 @@ class RuleTester {
      * @param {Function} rule The rule to test.
      * @param {{
      *   valid: (ValidTestCase | string)[],
-     *   invalid: InvalidTestCase[]
+     *   invalid: InvalidTestCase[],
+     *   linterMapDevOnly?: Map<string | undefined, Linter>
      * }} test The collection of tests to run.
+     * test.linterMapDevOnly is for testing purpose
```

Comment:
There is one unit test that need access to `linter` to verify the `parser`.
Can't think of a better way to inject the `linter`...

Reply:
It's doable to have `linterMap` to be a class property.
But I think it would be better not to do so, because it's used in `run` method only.
An additional method parameter is better than a class property in my point of view.

Reply:
Removing the instance field sounds nice.
I like using `Symbol("linterMapDevOnly")` in that case. We don't contain the symbol in public API and use it only in tests.

Reply:
For example, similarly [`lib/linter/linter.js` exposes `getLinterInternalSlots()` function for tests](https://github.com/eslint/eslint/blob/b7d79b12eb55f220f86367d52f9f5a2f503fd4c3/lib/linter/linter.js#L1456-L1463).